### PR TITLE
Features/Wallet Connection & User Role

### PR DESCRIPTION
**Pull request by JosephRana11**
Summary

- Integrated CIP-30 Wallet Connection: Enabled user signup/login via wallet connection.
- CIP-8 Verification: Added utility function for verifying signed data according to CIP-8 standards.
- Implemented JWT Authentication: Secured API with JSON Web Tokens.
- Added User Model ( User Roles -  Normal User and Admin.)
- New User Logic: Automatically creates a new agent with empty behavior and template upon first login.
- My Agent: Added "My Agent" endpoint and page for normal user interactions.
- Updated fronted agent view to hide/show agent manual trigger and run commands according to User Role/ Agent       Ownership
- Added test cases for User Login / User Logout.

Additional Updates

- Styling Updates: Enhanced the Drew Directory and Governance Action Directory to align with existing page styles.
- Code Cleanup: Removed unused components from the front-end.
- Skipped previous outdated test cases related to Agents and Templates

### CODE REVIEW

File: api/backend/app/controllers/agent_router.py [hunk 1]
```diff
@@ -41,8 +43,8 @@ async def get_agent(self, agent_id: str):
         return agent
 
     @put("/agents/{agent_id}", status_code=HTTPStatus.OK)
-    async def update_agent(self, agent_id: str, agent_data: AgentUpdateDTO):
-        updated_agent = await self.agent_service.update_agent(agent_id, agent_data)
+    async def update_agent(self, agent_id: str, agent_data: AgentUpdateDTO, user: dict = Depends(verify_cookie)):
```
Comment (mesudip) [inline api/backend/app/controllers/agent_router.py:46]: use TypedDict for user

File: api/backend/app/controllers/auth_router.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+class AuthRouter(Routable):
+
+    def __init__(self, user_service: UserService = UserService(), *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user_service = user_service
+        current_environment = os.environ.get("NODE_ENV")
+        self.secure = True if current_environment == "production" else False
+        self.domain = "api.agents.cardanoapi.io" if current_environment == "production" else "localhost"
+
+    @post("/login")
```
Comment (mesudip) [inline api/backend/app/controllers/auth_router.py:22]: set base path for all auth endpoint to start with '/auth'

File: api/backend/app/controllers/auth_router.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+from backend.app.models.user.user_dto import SignatureDataDto, UserCreateDto
+import os
+
+
+class AuthRouter(Routable):
+
+    def __init__(self, user_service: UserService = UserService(), *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user_service = user_service
+        current_environment = os.environ.get("NODE_ENV")
```
Comment (mesudip) [inline api/backend/app/controllers/auth_router.py:18]: use env variable name APP_ENV

Comment (mesudip) [inline api/backend/app/controllers/auth_router.py:18]: Use Pydantic Base Settings for configs like in this example : https://github.com/bettercollected/bettercollected/tree/develop/backend/backend/config

File: api/backend/app/services/agent_service.py [hunk 4]
```diff
@@ -72,22 +74,37 @@ async def get_agent(self, agent_id: str) -> AgentResponseWithWalletDetails:
...
+    async def update_agent(
+        self, agent_id: str, agent_data: AgentUpdateDTO, userAddress: str
+    ) -> AgentResponseWithAgentConfigurations:
+
         existing_agent = await self.agent_repository.retrieve_agent(agent_id)
         self.raise_exception_if_agent_not_found(existing_agent)
+        await self.is_authorized(userAddress, existing_agent)
+
+        user_is_super_admin = await self.user_reposiotry.is_super_admin(userAddress)
+        if existing_agent.user_address != userAddress and user_is_super_admin == False:
```
Comment (mesudip) [inline api/backend/app/services/agent_service.py:86]: this check is done twice in self.is_authorized and here.

**Review by mesudip — CHANGES_REQUESTED** [review]

File: agent-manager/prisma/schema.prisma [hunk 5]
```diff
@@ -1,5 +1,7 @@
 generator client {
-  provider = "prisma-client-js"
+  provider             = "prisma-client-py"
```
[the remark this answers is not in the document]
Comment (JosephRana11) [reply agent-manager/prisma/schema.prisma]: Reverted back to previous settings.

File: agent-manager/prisma/schema.prisma [hunk 6]
```diff
@@ -76,3 +82,12 @@ enum TriggerType {
...
   EVENT
 }
+
+model User {
+  address    String     @id
+  created_at DateTime
+  updated_at DateTime
+  deleted_at DateTime?
+  Agent      Agent[]
+  Template   Template[]
```
[the remark this answers is not in the document]
Comment (JosephRana11) [reply agent-manager/prisma/schema.prisma:90]: Synced Changes.

File: api/backend/app/controllers/agent_router.py (hunk 1 above)
Comment (JosephRana11) [reply api/backend/app/controllers/agent_router.py]: Created Typed Dict Model for User and replaced previous dict accordingly.

File: api/backend/app/controllers/auth_router.py (hunk 2 above)
Comment (JosephRana11) [reply api/backend/app/controllers/auth_router.py:20]: Set base path '/auth'  in both frontend and backend.

File: api/backend/app/controllers/auth_router.py (hunk 3 above)
Comment (JosephRana11) [reply api/backend/app/controllers/auth_router.py]: updated variable.

Comment (JosephRana11) [reply api/backend/app/controllers/auth_router.py]: Implemented API settings class using Pydantic base settings

File: api/backend/app/services/agent_service.py (hunk 4 above)
Comment (JosephRana11) [reply api/backend/app/services/agent_service.py]: Removed repeated code.

File: api/backend/config/api_settings.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+from pydantic import BaseSettings
+import os
+
+
+class APISettings(BaseSettings):
```
Comment (mesudip) [inline api/backend/config/api_settings.py:5]: Use Base settings feature. `os.environ.get` is not necessary.

**Review by mesudip — CHANGES_REQUESTED** [review]

**Review by mesudip — APPROVED** [review]

**Pull request merged**